src/components/organize/TagManager/components/TagSelect/index.tsx

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable jsx-a11y/no-autofocus */
+import { useAutocomplete } from '@material-ui/lab';
+import { useIntl } from 'react-intl';
+import { useState } from 'react';
+import { Box, TextField } from '@material-ui/core';
+
+import { groupTags } from '../../utils';
+import TagDialog from '../TagDialog';
+import TagSelectList from './TagSelectList';
+import ValueTagForm from './ValueTagForm';
+import { EditTag, NewTag } from '../../types';
+import { ZetkinTag, ZetkinTagGroup } from 'types/zetkin';
+
+const TagSelect: React.FunctionComponent<{
+  disableEditTags?: boolean;
+  disabledTags: ZetkinTag[];
+  groups: ZetkinTagGroup[];
+  onClose: () => void;
+  onCreateTag: (tag: NewTag) => Promise<ZetkinTag>;
+  onEditTag: (tag: EditTag) => void;
+  onSelect: (tag: ZetkinTag) => void;
+  tags: ZetkinTag[];
+}> = ({
+  disableEditTags,
+  disabledTags,
+  groups,
+  onClose,
+  onCreateTag,
+  onEditTag,
+  onSelect,
+  tags,
+}) => {
+  const intl = useIntl();
+
+  const [pendingTag, setPendingTag] = useState<ZetkinTag | null>(null);
+  const [inputValue, setInputValue] = useState('');
+  const [pendingValue, setPendingValue] = useState<number | string | null>(
+    null
+  );
+
+  const [tagToEdit, setTagToEdit] = useState<
+    ZetkinTag | Pick<ZetkinTag, 'title'> | undefined
+  >(undefined);
+
+  const { getInputProps, getListboxProps, getRootProps, groupedOptions } =
+    useAutocomplete({
+      getOptionLabel: (option) => option.title,
+      inputValue: inputValue,
+      open: true,
+      options: tags,
+    });
+
+  const groupedFilteredTags = groupTags(
+    groupedOptions,
+    intl.formatMessage({
+      id: 'misc.tags.tagManager.ungroupedHeader',
+    })
+  );
+
+  const handleSubmit = () => {
+    if (pendingTag) {
```

Comment:
Also I think this would be nicer to have inline. That's just personal preference. Up to you!

Reply:
Oh funny enough, it actually is inline

Reply:
It's used in two places, which is why I made it a separate function.